ui/DesignSystem/Component/SearchModal.svelte

```diff
@@ -0,0 +1,155 @@
+<script>
```

Comment:
```suggestion
<script lang="ts">
```

Now that we have the support for it, I'd expect every new component to be typescriped.

Reply:
I'm as eager as you are to get real ts into the components - capturing that work in #779